In [1]:
from sklearn.datasets import load_iris
import pandas as pd

iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)
df['species'] = iris.target
print(df.head())


   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   species  
0        0  
1        0  
2        0  
3        0  
4        0  


### Q1. How many rows and columns are present in the Iris dataset?

In [2]:
print(df.shape)

(150, 5)


### Q2. How many unique species exist in the dataset?

In [3]:
# Number of unique species
print(df['species'].nunique())

# Unique species labels
print(df['species'].unique())

3
[0 1 2]


### Q3. What is the average sepal length?

### Q4. How many flowers have a petal width > 1.5 cm?

### Q5. Are there any missing values in the dataset?

### Q6. Which species has the largest average petal length?

### Q7. Find the min, max, and mean of sepal width.

### Q8. How many samples have sepal_length > 6 AND petal_length > 5?

### Q9. Compute the correlation between sepal_length and petal_length.

### Q10. For each species, find the median petal_width.